src/ui.message.js

```diff
@@ -75,13 +75,8 @@ var _ = Mavo.UI.Message = $.Class({
 	},
 
 	close: function(resolve) {
-		if (this.options.style) {
-			var promise = this.options.style.transition === "none" ? Promise.resolve() : $.transition(this.element, {opacity: 0});
-		} 
-		else {
-			var promise = Promise.resolve();
-		}
-		promise.then(() => {
+		var closed = this.options.style && this.options.style.transition !== "none" ?  $.transition(this.element, {opacity: 0}) : Promise.resolve();
+		closed.then(() => {
```

Comment:
If you set `transition` to `none`, `transitionDuration` becomes `0s` anyway, so you can simplify this a lot by just setting the duration to whatever the computed style of `transitionDuration` is (converted to a number of milliseconds).
Relevant: https://blissfuljs.com/docs.html#fn-transition

Reply:
I tried: 
```
$.transition(this.element, {opacity: 0, duration: this.element.style.transitionDuration})
.then(() => {
	$.remove(this.element);
	this.closed.resolve(resolve);
});
```
But it appears to animate even when duration is 0.

Reply:
Please note "(converted to a number of milliseconds)." in my message above. You specified a string like "0s" where a number was expected.
Also, you put the duration in the wrong place. Please look at the documentation I linked to above.
Also, better use the computed style, not the inline style, in case the user has specified something different in their CSS.

Reply:
The animation still shows up (flashes) with this: 
```
$.transition(this.element, {opacity: 0}, 0)
.then(() => {
	$.remove(this.element);
	this.closed.resolve(resolve);
});
```

Reply:
Aha! That's because there's a bug in Bliss, right here: https://github.com/LeaVerou/bliss/blob/master/bliss.shy.js#L513

Better fix the bug in its source than work around it. That's why we use our own libraries whenever possible :)

Reply:
Do you want me to fix the bug in Bliss?

Reply:
Yup!

FYI Bliss was originally a part of Mavo, but was released separately before it, to solicit more contributions from the community.

Reply:
I updated the Bliss to `(duration !== undefined && duration !== null ? duration : 400) + "ms"` but it still flashes. (I replaced the bliss.shy.min.js in my local Mavo directory with the new Bliss). Any ideas why it's not working?

Reply:
Wouldn't you use 0 instead of 400?

Reply:
You don't need to take `null` into account. Catering for `undefined` is enough.

Or just `duration || duration === 0`